lib/src/screenshots.dart

```diff
@@ -0,0 +1,327 @@
+  }
+
+  final thumbnailProblems = await _generateThumbnails(
+      originalPath,
+      path.join(tempDir, webpPath),
+      path.join(tempDir, webpThumbnailPath),
+      path.join(tempDir, pngThumbnailPath));
+  if (thumbnailProblems.isNotEmpty) {
+    return ScreenshotResult.failed(thumbnailProblems);
+  }
+
+  return ScreenshotResult.success(
+      ProcessedScreenshot(e.path, e.description,
+          webpImage: webpPath,
+          webpThumbnail: webpThumbnailPath,
+          pngThumbnail: pngThumbnailPath),
+      tempDir);
+}
+
+/// Generates a WebP screenshot given the original image at [originalPath] and
+/// writes it to [webpPath].
+///
+/// Returns a list of problems if generating the screenshot did not succeed.
+/// On success, an empty list is returned.
+///
+/// The WebP screenshot can be either static or animated.
+///
+/// If the original image format is already WebP, the screenshot will simply be
+/// copied over. The `webpinfo` tool is used for determining this.
+/// Otherwise, a conversion is tried, using the `cwebp` tool which can handle
+/// PNG, JPEG, and TIFF as input. For GIF files the `gif2webp` conversion tool
+/// is used.
+///
+/// For any other input format the screenshot generation will fail. It will also
+/// fail if any of the mentioned tools are not available on the command line and
+/// an error message will be written to stderr.
+Future<List<String>> _generateWebpScreenshot(
+  String originalPath,
+  String webpPath,
+) async {
+  final problems = <String>[];
+  final copyIfWebpProblems = await _copyIfAlreadyWebp(originalPath, webpPath);
+  if (copyIfWebpProblems.isEmpty) {
+    return [];
+  }
+  problems.addAll(copyIfWebpProblems);
+
+  final convertWithCwebpProblems =
+      await _convertWithCWebp(originalPath, webpPath);
+  if (convertWithCwebpProblems.isEmpty) {
+    return [];
+  }
+  problems.addAll(convertWithCwebpProblems);
+
+  final gif2webpProblems = await _convertGifToWebp(originalPath, webpPath);
+  if (gif2webpProblems.isEmpty) {
+    return [];
+  }
+  problems.addAll(gif2webpProblems);
+  problems.add(
+      'Generating webp image for $originalPath failed due to invalid input');
+  return problems;
+}
+
+Future<List<String>> _copyIfAlreadyWebp(
+    String originalPath, String webpPath) async {
+  final infoResult = await _checkedRunProc(['webpinfo', originalPath]);
+  if (infoResult.exitCode == 0) {
+    await File(originalPath).copy(webpPath);
+    return [];
+  }
+  return [
+    '$originalPath: Tried interpreting screenshot as WebP. Failed with exitcode ${infoResult.exitCode}.'
+  ];
+}
+
+Future<List<String>> _convertWithCWebp(
+    String originalPath, String webpPath) async {
+  final cwebpResult =
+      await _checkedRunProc(['cwebp', originalPath, '-o', webpPath]);
+  if (cwebpResult.exitCode == 0) {
+    return [];
+  }
+  return [
+    '$originalPath: Converting screenshot with `cwebp` failed with exitcode ${cwebpResult.exitCode}.'
```

Comment:
```suggestion
    '`$originalPath`: Converting screenshot with `cwebp "$originalPath" -o result.webp` failed with _exit code_ `${cwebpResult.exitCode}`.'
```

If we use some backticks we get nice code formatting in markdown.

And if we provide the command, then advanced users can hopefully reproduce locally.

This is markdown, right?

Formatting advice regarding use of backticks applies to all these problems.

I think "exit code" is two words, or maybe "exit-code"? Alternatively we can write "... failed, exiting ${cwebpResult.exitCode}" -- but some people probably won't understand that.

Reply:
Done.